invenio_rdm_records/fixtures/tasks.py

```diff
@@ -27,3 +28,49 @@ def create_demo_record(data):
     service = current_rdm_records.records_service
     draft = service.create(data=data, identity=system_identity)
     service.publish(id_=draft.id, identity=system_identity)
+
+
+@shared_task(ignore_result=True)
+def update_expired_embargos():
+    """Release expired embargoes every midnight."""
+
+    def lift_embargo_from_draft(service, rdm_record):
+        """Lifts embargo from draft."""
+        record_draft = service.edit(rdm_record["id"], system_identity)
+        draft = record_draft._record
+        if draft.access.embargo.lift():
+            draft.access.protection.record = "public"
+            draft.access.protection.files = "public"
+        draft.commit()
+        return draft
+
+    def lift_embargo_from_record(rdm_record):
+        """Lifts embargo from record."""
+        if rdm_record.access.embargo.lift():
+            rdm_record.access.protection.record = "public"
+            rdm_record.access.protection.files = "public"
+            rdm_record.commit()
+
+    service = current_rdm_records.records_service
+    pid_type = "recid"
+    embargoed_q = "access.embargo.active:true"
+    restricted_records = service.scan(identity=system_identity, q=embargoed_q)
+    for restricted_record in restricted_records.to_dict()["hits"]["hits"]:
+        record = service.resolve_pid(
+            id_=restricted_record["id"],
+            pid_type=pid_type,
+            identity=system_identity
+        )
+        rdm_record = record._record
+        if arrow.utcnow().datetime > rdm_record.access.embargo.until:
+            if rdm_record.has_draft:
+                # Updates existing draft and existing record
+                lift_embargo_from_record(rdm_record)
+                draft = lift_embargo_from_draft(service, rdm_record)
+                service.indexer.index(draft)
+            else:
+                # Lifts the embargo by creating a new drat and publishing it
+                draft = lift_embargo_from_draft(service, rdm_record)
+                service.publish(draft["id"], system_identity)
+
+    db.session.commit()
```

Comment:
Should not be needed if we interact fully against the service layer.

Reply:
Yes, removed with the new approach